presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java

```diff
@@ -85,19 +91,24 @@ private synchronized void addPartition(TupleDomain<String> tupleDomain)
         // See the comment at TupleDomain::columnWiseUnion() for more details.
         partitions.add(tupleDomain);
         if (partitions.size() == partitionCount) {
-            Map<Symbol, Domain> result = convertTupleDomain(TupleDomain.columnWiseUnion(partitions));
+            TupleDomain<String> result = TupleDomain.columnWiseUnion(partitions);
             // No more partitions are left to be processed.
             resultFuture.set(result);
         }
     }
 
-    private Map<Symbol, Domain> convertTupleDomain(TupleDomain<String> result)
+    private Map<Symbol, Domain> convertTupleDomainForLocalFilters(TupleDomain<String> result)
     {
         if (result.isNone()) {
             // One of the join build symbols has no non-null values, therefore no symbols can match predicate
-            return buildChannels.keySet().stream()
-                    .flatMap(filterId -> probeSymbols.get(filterId).stream())
-                    .collect(toImmutableMap(identity(), probeSymbol -> Domain.none(types.get(probeSymbol))));
+            ImmutableMap.Builder<Symbol, Domain> builder = ImmutableMap.builder();
+            for (Map.Entry<String, Type> entry : filterBuildTypes.entrySet()) {
```

Comment:
this can be streamlined:
```
probeSymbols.entrySet().stream()
  .collect(toImmutableMap(
    entry -> entry.getValue(),
    entry -> filterBuildTypes.get(entry.getKey());
```

Reply:
`probeSymbols` is `Multimap<String, Symbol>` so this doesn't work directly.